src/System.Memory/tests/BuffersExtensions/BuffersExtensionsTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class BuffersExtensionsTests
+    {
+        [Fact]
+        public void WritingToSingleSegmentBuffer()
+        {
+            IBufferWriter<byte> bufferWriter = new TestBufferWriterSingleSegment();
+            bufferWriter.Write(Encoding.UTF8.GetBytes("Hello"));
+            bufferWriter.Write(Encoding.UTF8.GetBytes(" World!"));
+            Assert.Equal("Hello World!", bufferWriter.ToString());
+        }
+
+        [Fact]
+        public void WritingToMultiSegmentBuffer()
+        {
+            IBufferWriter<byte> bufferWriter = new TestBufferWriterMultiSegment();
+            bufferWriter.Write(Encoding.UTF8.GetBytes("Hello"));
+            bufferWriter.Write(Encoding.UTF8.GetBytes(" World!"));
+            Assert.Equal("Hello World!", bufferWriter.ToString());
+        }
+
+        private class TestBufferWriterSingleSegment : IBufferWriter<byte>
+        {
+            private byte[] _buffer = new byte[1000];
+            private int _written = 0;
+
+            public void Advance(int bytes)
+            {
+                _written += bytes;
+            }
+
+            public Memory<byte> GetMemory(int sizeHint  = 0) => _buffer.AsMemory().Slice(_written);
+
+            public Span<byte> GetSpan(int sizeHint) => _buffer.AsSpan().Slice(_written);
```

Comment:
nit: ` _buffer.AsSpan().Slice(_written)` can be written more concisely as  `_buffer.AsSpan(_written)`